pkg/cmd/release/deploy/deploy.go

```diff
@@ -0,0 +1,1194 @@
+	var controlType variables.ControlType
+	if displaySettings != nil && displaySettings.ControlType != "" {
+		controlType = displaySettings.ControlType
+	} else { // infer the control type based on other flags
+		// The shape of the data model allows for the possibility of a sensitive multi-line or sensitive combo-box
+		// variable. However, the web portal doesn't implement any of these, the only sensitive thing it supports
+		// is single-line text, so we can simplify our logic here.
+		if variableType == "Sensitive" || isSensitive {
+			// From comment in server:
+			// variable.IsSensitive is Kept for backwards compatibility. New way is to use variable.Type=VariableType.Sensitive
+			controlType = variables.ControlTypeSensitive
+		} else {
+			controlType = variables.ControlTypeSingleLineText
+		}
+	}
+
+	switch controlType {
+	case variables.ControlTypeSingleLineText, "": // if control type is not explicitly set it means single line text.
+		var response string
+		err := asker(&survey.Input{
+			Message: message,
+			Default: defaultValue,
+		}, &response, askOpt)
+		return response, err
+
+	case variables.ControlTypeSensitive:
+		var response string
+		err := asker(&survey.Password{
+			Message: message,
+		}, &response, askOpt)
+		return response, err
+
+	case variables.ControlTypeMultiLineText: // not clear if the server ever does this
+		var response string
+		err := asker(&surveyext.OctoEditor{
+			Editor: &survey.Editor{
+				Message:  "message",
+				FileName: "*.txt",
+			},
+			Optional: !isRequired}, &response)
+		return response, err
+
+	case variables.ControlTypeSelect:
+		if displaySettings == nil {
+			return "", cliErrors.NewArgumentNullOrEmptyError("displaySettings") // select needs actual display settings
+		}
+		reverseLookup := make(map[string]string, len(displaySettings.SelectOptions))
+		optionStrings := make([]string, 0, len(displaySettings.SelectOptions))
+		displayNameForDefaultValue := ""
+		for _, v := range displaySettings.SelectOptions {
+			if v.Value == defaultValue {
+				displayNameForDefaultValue = v.DisplayName
+			}
+			optionStrings = append(optionStrings, v.DisplayName)
+			reverseLookup[v.DisplayName] = v.Value
+		}
+		var response string
+		err := asker(&survey.Select{
+			Message: message,
+			Default: displayNameForDefaultValue,
+			Options: optionStrings,
+		}, &response, askOpt)
+		if err != nil {
+			return "", err
+		}
+		return reverseLookup[response], nil
+
+	case variables.ControlTypeCheckbox:
+		// if the server didn't specifically set a default value of True then default to No
+		defTrueFalse := "False"
+		if b, err := strconv.ParseBool(defaultValue); err == nil && b {
+			defTrueFalse = "True"
+		}
+		var response string
+		err := asker(&survey.Select{
+			Message: message,
+			Default: defTrueFalse,
+			Options: []string{"True", "False"}, // Yes/No would read more nicely, but doesn't fit well with cmdline which expects True/False
+		}, &response, askOpt)
```

Comment:
why did we chose to use select; over `survery.Confirm` and covert the response bool to a string?

Reply:
No strong reason. This whole thing still needs a 'ux' pass with JB and shannon, we might yet change some of this kind of stuff.

I chose a select with a true/false because it feels "easier", you can just sit on the arrow keys and enter rather than having to type 'y' on the keyboard. Also (perhaps more importantly) because the command line flag requires you to specify "true" or "false", whereas confirm is always y/n, so it's - in a small way - nice to present true/false here as well